TradingOnTalentURAP/degree_labelling.py

```diff
@@ -0,0 +1,103 @@
+import pandas as pd
+
+# Labelling Function
+
+def assign_labels(df, direct_keywords, direct_star_keywords, exclude_keywords, tangential_keywords):
+
+    df_direct = df[df.Degree.str.contains('|'.join(direct_keywords), case = False)]
+
+    df_direct = df_direct[~df_direct.Degree.str.contains('|'.join(exclude_keywords), case = False)]
+    
+    df_without_exclude = df[~df.Degree.str.contains('|'.join(set(df_direct.Degree)), case = False, regex = False)]
+    
+    df_tangential = df_without_exclude[df_without_exclude.Degree.str.contains('|'.join(tangential_keywords), case = False)]
+    
+    df_direct_star = df_without_exclude[df_without_exclude.Degree.str.contains('|'.join(direct_star_keywords), case = False)]
+    
+    df_labels = []
+    for index, row in df.iterrows():
+        if row['Degree'] in set(df_direct.Degree):
+            df_labels.append('direct')
+        elif row['Degree'] in set(df_tangential.Degree):
+            df_labels.append('indirect')
+        elif row['Degree'] in set(df_direct_star.Degree):
+            df_labels.append('direct*')
+        else:
+            df_labels.append("None")
+        
+    df['Label'] = df_labels
+    
+    return df
+
+ # Load in all dataframes
+
+dfs = {}
+universities = ['anna university.csv', 
```

Comment:
python's glob module would allow you to avoid listing all files--they may change in the future:
import glob
import os
files_list = glob.glob(source + os.path.sep + '*.csv')
where source is the directory containing the files that was passed into your program and parsed via argparse (see comment below). In this case, you would run your script approximately like:

python degree_labelling.py /Users/jacquelinewood/Documents/URAP/Universities_Unlabelled